### PR TITLE
update the SENSEI in situ coupling for SENSEI v4.0.0

### PR DESCRIPTION
**Pull request by burlen**
## Summary
In this release of SENSEI, an install of VTK is no longer required. To compile AMReX w/ SENSEI use:

```cmake
-DAMReX_SENSEI=ON -DSENSEI_DIR=<path to SENSEI install>/lib64/cmake
```

Tested with AMReX CMake build system, a validation using GNUMakefiles will come later.

## Additional background
None

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by burlen** [comment]
@kwryankrattiger Can you update the docker image used for regression tests here to include the latest install of SENSEI?

**Comment by kwryankrattiger** [comment]
@burlen I am working on the new image, I will update you when it is pushed.

**Comment by kwryankrattiger** [comment]
@burlen Sorry for the delay on this. The required changes here are

* New image: `senseiinsitu/ci:fedora35-amrex-20220613`
* New location for sensei in  CMAKE_PREFIX_PATH; `/root/install/sensei/v4.0.0/lib/cmake`

I tested this locally and didn't have an issue, but if it gives you trouble let me know.

Reference this PR for the changes made to the container #2834 .

**Comment by ax3l** [comment]
Not sure if you already did this, but generally can you please update CMake to the latest version in the existing (and new/4.0) container? :)

It's currently at `3.19.7` but I like to bump WarpX to 3.20+ (or even newer): https://github.com/ECP-WarpX/WarpX/pull/3196

**Comment by burlen** [comment]
@ax3l I believe we can update to a newer cmake without issue, I'm locally using 3.22.2 and have no issues.

Upgrading cmake will go in some other pr than this one. That PR will likely include a few other changes. #2834 uses the sensei development branch. Instead we should use an official sensei release so that amrex codes are not exposed to our unstable development codes. There have been enough build related bug fixes to sensei in the past few weeks that we'll need a 4.0.1 release. 

@kwryankrattiger can you help upgrade cmake, and pin the amrex testing on the sensei 4.0.1 release ?

**Comment by kwryankrattiger** [comment]
@ax3l When I built the container I updated the OS version from fedora 33 to 35 so it should be using cmake 3.22 now. This is also consistent with the CI containers SENSEI is using for testing.

**Comment by kwryankrattiger** [comment]
@burlen AMRex sensei CI is pinned to what is effectively v4.0.1, it uses a containerized sensei that is fixed on the working develop as of 2022-06-13 as denoted by the container, so it will not change unless we update the container AMReX is using for sensei CI.

**Comment by burlen** [comment]
@kwryankrattiger Oh, OK, my bad.

**Comment by burlen** [comment]
@kwryankrattiger can you confirm that the workflow get updated with the path to the new container? I guess that's probably why Axel is still having the issue. It is also the case that we need to update the adaptor code in warpx for the new sensei 4 api's. I can start working on it but it depends on this pr

**Comment by kwryankrattiger** [comment]
@burlen I want to say @c-wetterer-nelson is working on the warpx update for sensei v4. 

I didn't update the amrex sensei CI yet since it would make the action red without your changes. If you can update `.github/workflows/sensei.yml` in this PR to set the `jobs:test_sensei:container:image` to `senseiinsitu/ci:fedora35-amrex-20220613` and update the `jobs:test_sensei:env:CMAKE_PREFIX_PATH` to point at `/root/install/sensei/v4.0.0/lib/cmake` then the CI for sensei will pass for this PR.

File: .github/workflows/sensei.yml [hunk 1]
```diff
@@ -17,9 +17,9 @@ jobs:
       CC: clang
       CXXFLAGS: "-Werror -Wshadow -Woverloaded-virtual -Wunreachable-code -fno-operator-names"
       CMAKE_GENERATOR: Ninja
-      CMAKE_PREFIX_PATH: /root/install/sensei/develop/lib/cmake
+      CMAKE_PREFIX_PATH: /root/install/sensei/v4.0.0/lib/cmake
```
Comment (kwryankrattiger) [inline .github/workflows/sensei.yml]: @burlen My apologies, it is `lib64` not `lib` now for some reason.

Comment (burlen) [reply .github/workflows/sensei.yml]: no worries, I knew about that, it's because we've adopted the GNUInstallDirs conventions from CMake.

**Comment by burlen** [comment]
@WeiqunZhang would someone on the amrex team be willing to review this PR? I think it is ready

**Comment by WeiqunZhang** [comment]
LGTM.  Could you also update `Docs/sphinx_documentation/source/Visualization.rst`? The build instruction for CMake uses `lib`, whereas the PR descriptor has `lib64`.  From the discussion above, looks like it should be updated.

**Comment by burlen** [comment]
OK, `lib` vs `lib64` is determined by CMake based on OS at configure time. I added a note about this to the docs

File: Src/Extern/SENSEI/AMReX_AmrMeshDataAdaptor.cpp [hunk 2]
```diff
@@ -711,18 +711,18 @@ int AmrMeshDataAdaptor::AddGhostCellsArray(vtkDataObject* mesh,
...
+            svtkUnsignedCharArray *ga = svtkUnsignedCharArray::New();
+            ga->SetName("svtkGhostType");
             ga->SetArray(mask[j], nCells, 0);
             blockMesh->GetCellData()->AddArray(ga);
             ga->Delete();
 
             // for debug can visualize the ghost cells
             // FIXME -- a bug in Catalyst ignores internal ghost zones
-            // when using the VTK writrer. Until that bug gets fixed, one
+            // when using the SVTK writrer. Until that bug gets fixed, one
```
Comment (ax3l) [inline Src/Extern/SENSEI/AMReX_AmrMeshDataAdaptor.cpp]:
```suggestion
            // when using the SVTK writer. Until that bug gets fixed, one
```

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**